components/arm/fake/fake.go

```diff
@@ -125,6 +131,20 @@ type Arm struct {
 	model      referenceframe.Model
 }
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (a *Arm) UpdateAction(c *config.Component) config.UpdateActionType {
+	var err error
+	if _, ok := c.ConvertedAttributes.(*AttrConfig); ok {
+		a.model, err = buildModel(*c)
+		if err != nil {
+			a.logger.Fatalln("could not build model:", err.Error())
```

Comment:
So you cannot do fatal here. It's going to crash the program. UpdateAction does not support ctx/error meaning you can't *fully* use it for updating. Its actual intended usage is to check who kind of update should happen, not to process the config.

You may need to return config.Rebuild.

cc: @cheukt example of where this model isn't working too well